src/gui/ocssharejob.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) by Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "ocssharejob.h"
+#include "networkjobs.h"
+#include "account.h"
+#include "json.h"
+
+#include <QBuffer>
+
+namespace OCC {
+
+OcsShareJob::OcsShareJob(AccountPtr account, QObject* parent)
+: OCSJob(account, parent)
+{
+    setUrl(Account::concatUrlPath(account->url(), QString("ocs/v1.php/apps/files_sharing/api/v1/shares")));
+}
+
+OcsShareJob::OcsShareJob(int shareId, AccountPtr account, QObject* parent)
+: OCSJob(account, parent)
+{
+    setUrl(Account::concatUrlPath(account->url(), QString("ocs/v1.php/apps/files_sharing/api/v1/shares/%1").arg(shareId)));
+}
+
+void OcsShareJob::getShares(const QString &path)
+{
+    setVerb("GET");
+    
+    QList<QPair<QString, QString> > getParams;
+    getParams.append(qMakePair(QString::fromLatin1("path"), path));
+    setGetParams(getParams);
+
```

Comment:
I think an API that provides a method `addParam( QPair())` instead of the `setGetParams(QList())` would be nicer because it would save to set up the List and all the pairing here. Could we also use just an `addParam()`  rather than addGetParam() and `addPostParam()`?